server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/scm/ScmInfoRepositoryImpl.java

```diff
@@ -87,6 +103,24 @@ private static ScmInfo getScmInfoFromReport(Component file, ScannerReport.Change
     return new ReportScmInfo(changesets);
   }
 
+  private ScmInfo generatedScmInfo(Component file) {
+    ScmInfo dbInfo = scmInfoDbLoader.getScmInfoFromDb(file);
+    if (dbInfo == NoScmInfo.INSTANCE) {
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Remove this conditional structure or edit its code blocks so that they're not all the same. ([squid:S3923](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3923))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Fdm%2Fno_scm_info&issues=AWD56Pl1EFfjxkQcPmlT&open=AWD56Pl1EFfjxkQcPmlT)

[ISSUE_KEY]: AWD56Pl1EFfjxkQcPmlT